src/WebUI/Pages/Leaderboard.razor

```diff
@@ -0,0 +1,122 @@
+                </NotAuthorized>
+            </AuthorizeView>
+            <MudSpacer/>
+        </MudStack>
+        <MudButtonGroup OverrideStyles="false">
+            <MudButton
+                Color="Color.Error"
+                DisableElevation="true"
+                Variant="@(Filter == LeaderboardFilter.LastMonth ? Variant.Filled : Variant.Outlined)"
+                OnClick="@(() => HandleFilterChanged(LeaderboardFilter.LastMonth))">
+                Last Month
+            </MudButton>
+            <MudButton
+                Color="Color.Error"
+                DisableElevation="true"
+                Variant="@(Filter == LeaderboardFilter.LastYear ? Variant.Filled : Variant.Outlined)"
+                OnClick="@(() => HandleFilterChanged(LeaderboardFilter.LastYear))">
+                Last Year
+            </MudButton>
+            <MudButton
+                Color="Color.Error"
+                DisableElevation="true"
+                Variant="@(Filter == LeaderboardFilter.AllTime ? Variant.Filled : Variant.Outlined)"
+                OnClick="@(() => HandleFilterChanged(LeaderboardFilter.AllTime))">
+                All Time
+            </MudButton>
+        </MudButtonGroup>
+        <MudTable Items="@LeaderboardData" T="LeaderboardUserStats">
+            <HeaderContent>
+                <MudTh>Position</MudTh>
+                <MudTh>Name</MudTh>
+                <MudTh>Messages Sent</MudTh>
+            </HeaderContent>
+            <RowTemplate>
+                <MudTd DataLabel="Position">@context.Position</MudTd>
+                <MudTd DataLabel="Name">@context.Name</MudTd>
+                <MudTd DataLabel="Messages Sent">
+                    @switch (Filter)
+                    {
+                        case LeaderboardFilter.LastMonth:
+                            @context.LastMonth
+                            break;
+
+                        case LeaderboardFilter.LastYear:
+                            @context.LastYear
+                            break;
+
+                        case LeaderboardFilter.AllTime:
+                            @context.AllTime
+                            break;
+
+                        default:
+                            throw new ArgumentOutOfRangeException();
+                    }
+                </MudTd>
+            </RowTemplate>
+            <PagerContent>
+                <MudTablePager PageSizeOptions="new int[] { 10, 50 }"/>
+            </PagerContent>
+        </MudTable>
+    </MudStack>
+</div>
+
+@code {
+    private List<LeaderboardUserStats> LeaderboardData { get; set; } = null!;
+    private LeaderboardFilter Filter { get; set; }
+
+    protected override async Task OnInitializedAsync()
+    {
+        var result = await Http.GetFromJsonAsync<List<LeaderboardUserStats>>($"{Configuration.GetValue<string>("ApiBaseUrl")}/getLeaderboardStats");
```

Comment:
You don't need to raw dog this HTTP call - use the RulesGptApiClients that are autogenerated 

i.e. RulesGptClient.GetLeaderboardStatsAsync